server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/mapping/TextTransformations.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.engine.mapping;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.lsp.cobol.core.model.tree.CopyNode;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+
+import java.util.*;
+
+/**
+ * This value class represents the extended document hierarchy. It's possible to construct extended
+ * source based on this data.
+ */
+@Data
+@RequiredArgsConstructor
+public class TextTransformations {
+  public static final String REGEX_KEEP_NEW_LINES = "(?<=\\r?\\n)";
+  private final String text;
+  private final String uri;
+  private final Map<Range, TextTransformations> extensions = new HashMap<>();
+  private final Map<Range, String> replacements = new HashMap<>();
+  private final List<CopyNode> copyNodes = new ArrayList<>();
+
+  /**
+   * Apply all transformations and form resulting text
+   *
+   * @return text with all transformations
+   */
+  public String calculateExtendedText() {
+    StringBuilder eda = new StringBuilder();
+
+    LinkedList<Range> ranges = new LinkedList<>(extensions.keySet());
+    ranges.addAll(replacements.keySet());
+    ranges.sort(
+        Comparator.comparingInt((Range c) -> c.getStart().getLine())
+            .thenComparingInt(c -> c.getStart().getCharacter()));
+
+    String[] lines = text.split(REGEX_KEEP_NEW_LINES, -1);
```

Comment:
Sorry, I didn't follow this. Any reason to not split on lineSeparator but to use a look around regex?

Reply:
The reason is that some files contain only "\n" and some files contain "\r\n" so "\r" is optional